core/src/main/java/org/apache/accumulo/core/conf/ConfigSanityCheck.java

```diff
@@ -88,6 +91,12 @@ else if (!prop.getType().isValidFormat(value))
       if (key.equals(Property.CRYPTO_CIPHER_KEY_ALGORITHM_NAME.getKey())) {
         keyAlgorithm = Objects.requireNonNull(value);
       }
+      if (key.equals(Property.CRYPTO_SECRET_KEY_ENCRYPTION_STRATEGY_CLASS.getKey()) && !(value == null || value.equals("NullSecretKeyEncryptionStrategy"))) {
```

Comment:
Should this be `value.equals(NullSecretKeyEncryptionStrategy.class.getName()) || value.equals(NullSecretKeyEncryptionStrategy.class.getSimpleName())` instead of checking value by String literal?

(Similar comment below on String literal comparison for "NullCryptoModule")

Reply:
NullSecretKeyEncryptionStrategy and NullCryptoModule are both package private and nested within CryptoModuleFactory. I am not sure why, so I am hesitant to change them here. It is probably worth considering making both public and moving both out of CryptoModuleFactory in a future PR.

Reply:
Ah, I see. It probably doesn't matter much anyway.